File: app/src/main/java/com/kakacat/minitool/common/util/StringUtil.java
package com.kakacat.minitool.common.util;

import android.icu.text.SimpleDateFormat;

public class StringUtil {

    public static String byteToMegabyte(long bytes){
        String result;
        if(bytes > 1024 * 1024 * 1024){
            String temp = String.valueOf(bytes / (1024 * 1024 * 1024.0));
            result = temp.substring(0,temp.indexOf('.') + 2) + "G";
        }else{
            String temp = String.valueOf(bytes / (1024 * 1024.0));
            result = temp.substring(0,temp.indexOf('.') + 2) + "M";
        }
        return result;
    }

    public static String getDate(long time){
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        return sdf.format(time);
    }

    public static String byteToString(byte[] bytes,boolean addColon){
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < bytes.length; i++) {
            if (Integer.toHexString(0xFF & bytes[i]).length() == 1)
                sb.append("0").append(Integer.toHexString(0xFF & bytes[i]));
            else
                sb.append(Integer.toHexString(0xFF & bytes[i]));
            if(addColon)
                if(i != bytes.length - 1)
                    sb.append(':');
        }
        return sb.toString();
    }


}

File: app/src/main/java/com/kakacat/minitool/common/util/StringUtil.kt
package com.kakacat.minitool.common.util

import android.icu.text.SimpleDateFormat

object StringUtil {
    fun byteToMegabyte(bytes: Long): String {
        val result: String
        result = if (bytes > 1024 * 1024 * 1024) {
            val temp = (bytes / (1024 * 1024 * 1024.0)).toString()
            temp.substring(0, temp.indexOf('.') + 2) + "G"
        } else {
            val temp = (bytes / (1024 * 1024.0)).toString()
            temp.substring(0, temp.indexOf('.') + 2) + "M"
        }
        return result
    }

    @JvmStatic
    fun getDate(time: Long): String {
        val sdf = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
        return sdf.format(time)
    }

    @JvmStatic
    fun byteToString(bytes: ByteArray, addColon: Boolean): String {
        val sb = StringBuilder()
        for (i in bytes.indices) {
            if (Integer.toHexString(0xFF and bytes[i].toInt()).length == 1) sb.append("0").append(Integer.toHexString(0xFF and bytes[i].toInt())) else sb.append(Integer.toHexString(0xFF and bytes[i].toInt()))
            if (addColon) if (i != bytes.size - 1) sb.append(':')
        }
        return sb.toString()
    }
}